src/SessionSetup/ServiceDeployManager.cpp

```diff
@@ -449,78 +451,102 @@ ErrorMessageOr<void> ServiceDeployManager::CopyOrbitUserSpaceInstrumentationLibr
-                   &ServiceDeployManager::handleSocketError);
-  QObject::connect(&orbit_service_task_.value(), &orbit_ssh_qt::Task::finished, this,
-                   [](int exit_code) {
-                     ORBIT_LOG("The OrbitService Task finished with exit code: %d", exit_code);
-                   });
-  return outcome::success();
+  return command;
 }
 
-ErrorMessageOr<void> ServiceDeployManager::StartOrbitServicePrivileged(
-    const BareExecutableAndRootPasswordDeployment& config) {
+ErrorMessageOr<void> ServiceDeployManager::StartOrbitService(
+    const std::variant<SignedDebianPackageDeployment, BareExecutableAndRootPasswordDeployment>&
+        deployment_config) {
   ORBIT_CHECK(QThread::currentThread() == thread());
-  // TODO(antonrohr) Check whether the password was incorrect.
-  // There are multiple ways of doing this. the best way is probably to have a
-  // second task running before OrbitService that sets the SUID bit. It might be
-  // necessary to close stdin by sending EOF, since sudo would ask for trying to
-  // enter the password again. Another option is to use std err as soon as its
-  // implemented in OrbitSshQt::Task.
   emit statusMessage("Starting OrbitService on the remote instance...");
 
-  std::string task_string = "sudo --stdin /tmp/OrbitService";
-  if (absl::GetFlag(FLAGS_devmode)) {
-    task_string += " --devmode";
-  }
+  std::string task_string = GenStartOrbitServiceCommand(deployment_config);
   orbit_service_task_.emplace(&session_.value(), task_string);
 
-  orbit_service_task_->Write(absl::StrFormat("%s\n", config.root_password));
+  if (std::holds_alternative<BareExecutableAndRootPasswordDeployment>(deployment_config)) {
+    const auto& config = std::get<BareExecutableAndRootPasswordDeployment>(deployment_config);
+    orbit_service_task_->Write(absl::StrFormat("%s\n", config.root_password));
+    // TODO(antonrohr) Check whether the password was incorrect.
+    // There are multiple ways of doing this. the best way is probably to have a
+    // second task running before OrbitService that sets the SUID bit. It might be
+    // necessary to close stdin by sending EOF, since sudo would ask for trying to
+    // enter the password again. Another option is to use std err as soon as its
+    // implemented in OrbitSshQt::Task.
+  }
 
   orbit_qt_utils::EventLoop loop{};
   auto error_handler =
       ConnectErrorHandler(&loop, &orbit_service_task_.value(), &orbit_ssh_qt::Task::errorOccurred);
-  auto quit_handler =
-      ConnectQuitHandler(&loop, &orbit_service_task_.value(), &orbit_ssh_qt::Task::started);
   auto cancel_handler = ConnectCancelHandler(&loop, this);
+  QObject::connect(
+      &orbit_service_task_.value(), &orbit_ssh_qt::Task::finished, &loop, [&loop](int exit_code) {
+        // TODO(http://b/221369463): So far this is only reporting a rather generic exit code. This
+        // will be extended in a subsequent PR.
```

Comment:
I guess this is very temporary but I don't think the text should refer to "current PR" in TODOs, rather explain what is left to do.

Reply:
Hopefully it's explained better now.